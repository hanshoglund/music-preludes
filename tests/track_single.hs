let
    x = [ (0, c), (1, d), (2, e) ]^.track
    y = join $ [ (0, x), 
                (1.5,  up _P5 x), 
                (3.25, up _P8 x) ]^.track

in trackToScore (1/8) y
