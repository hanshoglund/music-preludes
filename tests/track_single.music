let
    x = [ (0, c)^.placed, (1, d)^.placed, (2, e)^.placed ]^.track
    y = join $ [ (0, x)^.placed, 
                (1.5,  up _P5 x)^.placed, 
                (3.25, up _P8 x)^.placed ]^.track

    trackToScore d = view score . map (view event . (\(t,x) -> (t >-> d,x)) . (view $ from placed)) . view placeds
    
in trackToScore (1/8) y
