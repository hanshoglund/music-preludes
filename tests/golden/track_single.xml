<?xml version='1.0' ?>
<score-partwise>
  <movement-title></movement-title>
  <identification>
    <creator type="composer"></creator>
  </identification>
  <part-list>
    <score-part id="P1">
      <part-name></part-name>
    </score-part>
  </part-list>
  <part id="P1">
    <measure number="1">
      <attributes>
        <divisions>768</divisions>
      </attributes>
      <attributes>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
      </attributes>
      <direction>
        <direction-type>
          <metronome>
            <beat-unit>whole</beat-unit>
            <per-minute>60</per-minute>
          </metronome>
        </direction-type>
      </direction>
      <attributes>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <attributes>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
      </attributes>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>C</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
    <measure number="2">
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>D</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>G</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
    <measure number="3">
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>E</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>A</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
    <measure number="4">
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>C</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>B</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest />
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
    <measure number="5">
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>D</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest />
        <duration>1024</duration>
        <voice>1</voice>
        <type>half</type>
        <time-modification>
          <actual-notes>3</actual-notes>
          <normal-notes>2</normal-notes>
        </time-modification>
        <notations>
          <ornaments />
          <technical />
          <articulations />
          <tuplet number="1" type="start" />
        </notations>
      </note>
      <note>
        <rest />
        <duration>512</duration>
        <voice>1</voice>
        <type>quarter</type>
        <time-modification>
          <actual-notes>3</actual-notes>
          <normal-notes>2</normal-notes>
        </time-modification>
      </note>
      <note>
        <rest />
        <duration>256</duration>
        <voice>1</voice>
        <type>eighth</type>
        <time-modification>
          <actual-notes>3</actual-notes>
          <normal-notes>2</normal-notes>
        </time-modification>
      </note>
      <note>
        <rest />
        <duration>128</duration>
        <voice>1</voice>
        <type>16th</type>
        <time-modification>
          <actual-notes>3</actual-notes>
          <normal-notes>2</normal-notes>
        </time-modification>
        <notations>
          <ornaments />
          <technical />
          <articulations />
          <tuplet number="1" type="stop" />
        </notations>
      </note>
    </measure>
    <measure number="6">
      <attributes>
        <time>
          <beats>3</beats>
          <beat-type>8</beat-type>
        </time>
      </attributes>
      <note>
        <rest />
        <duration>768</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>E</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
  </part>
</score-partwise>
